eng/common/TestResources/New-TestResources.ps1

```diff
@@ -92,6 +92,9 @@ param (
     [Parameter()]
     [switch] $SuppressVsoCommands = ($null -eq $env:SYSTEM_TEAMPROJECTID),
 
+    [Parameter()]
+    [switch] $UseUserCredentials,
```

Comment:
I'm not in love with this name, but I can't seem to think of one that works better.    "$NoServicePrincipal" and "$RunAsCurrentUser" are the best that I've got, and I think they're potentially confusing, too.

Reply:
Agreed RunAsCurrentUser sounds better but is also confusing since we still are using user information even without the new flag set.

Reply:
Maybe "$AssignCurrentUserCredentials"?    (still terrible)

Reply:
Don't we want to default this to `$true`?
```suggestion
    [switch] $UseUserCredentials = $true,
```
People can either pass `-UseUserCredentials:$false`, which is wonky, or we rename it so that the default works in a non-negative case.

Reply:
See https://github.com/Azure/azure-sdk-tools/pull/7580#discussion_r1465295760

Reply:
`UserAuth` `PassUserAuthToTest` `RunTestsAsCurrentUser`

I agree, I think we can come up with an alternative to `UseUser`.

Reply:
UserAuth works for me

Reply:
`$CreateResourcesForCurrentUser` ??

Reply:
The resources are already created with the current user credentials

Reply:
`$AuthorizeCurrentUser`?